sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1324,6 +1324,16 @@ object SQLConf {
     .intConf
     .createWithDefault(2)
 
+  val STATE_STORE_COMPRESSION_CODEC =
+    buildConf("spark.sql.streaming.stateStore.compression.codec")
+      .internal()
+      .doc("The codec used to compress delta and snapshot files generated by StateStore. " +
+        "By default, Spark provides four codecs: lz4, lzf, snappy, and zstd. You can also " +
+        "use fully qualified class names to specify the codec. Default codec is lz4.")
```

Comment:
Shall we add a test case for this assertion, `fully qualified class names`? In the new test case, it seems that we always use `getShortName`.

Reply:
Also, I'm wondering what happens when the invalid class name. It would be great if we can add a negative test case.

Reply:
Okay.

Reply:
For invalid codec name, `CompressionCodec.createCodec` will throw an `IllegalArgumentException`.